src/table.js

```diff
@@ -538,15 +543,105 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  const stringValue = typeof value === "string" ? value.trim() : value;
```

Comment:
The name `stringValue` is a little confusing here because if `value` is not a string, then `stringValue` is not a string, too. It’s kind of like `maybeTrimValue`? But I think perhaps we can avoid this local variable entirely by changing some of the logic below, and thereby only trim if needed; see comments below.